### PR TITLE
Fix offline datagen for chunked prefill

### PR DESCRIPTION
**Pull request by shanjiaz**
This PR should address the issue raised in [#279](https://github.com/vllm-project/speculators/issues/279). 

## Bakground
vLLM
1. Splits requests across multiple scheduler iterations
2. Reorders tokens internally
3. Mixes prefill and decode tokens in the same batch

The original implementation assumed tokens were concatenated in input order and sliced them incorrectly.

## Changes
  custom_worker.py:
  - Pass metadata as a dict(mapping request_ids to num_tokens on the request) from generator to worker
  - Query `input_batch.req_id_to_index` to determine actual token order
  - Return results as a dict mapping request_ids to hidden states instead of concatenated tensor

  vllm_hidden_states_generator.py:
  - Disable prefix caching in cache config
  - Track how many tokens have been computed per request to only capture prefill tokens (the original implementation aborted all requests after each iteration, causing partial prefills to be lost)
  - Stop after all prefill complete, abort remaining decode

  test_vllm_hidden_states.py:
  - Added `test_batch_vs_individual_consistency` regression test based on community member's script
  - Validates batch processing matches individual processing exactly

## Test plan & results
Tested with the script community member provided:
```
======================================================================
BATCH PROCESSING (all 8 at once)
======================================================================
[2026-02-09 16:17:02] INFO logging_utils.py:49:   Scheduler iteration 1 - total_tokens: 348
[2026-02-09 16:17:02] INFO logging_utils.py:49:     req_8_0: 42 tokens
[2026-02-09 16:17:02] INFO logging_utils.py:49:     req_8_1: 46 tokens
[2026-02-09 16:17:02] INFO logging_utils.py:49:     req_8_2: 43 tokens
[2026-02-09 16:17:02] INFO logging_utils.py:49:     req_8_3: 45 tokens
[2026-02-09 16:17:02] INFO logging_utils.py:49:     req_8_4: 37 tokens
[2026-02-09 16:17:02] INFO logging_utils.py:49:     req_8_5: 48 tokens
[2026-02-09 16:17:02] INFO logging_utils.py:49:     req_8_6: 42 tokens
[2026-02-09 16:17:02] INFO logging_utils.py:49:     req_8_7: 45 tokens
[2026-02-09 16:17:03] INFO logging_utils.py:49:   Aborting 8 requests after prefill complete
  Seq 0: OK
  Seq 1: OK
  Seq 2: OK
  Seq 3: OK
  Seq 4: OK
  Seq 5: OK
  Seq 6: OK
  Seq 7: OK

SUMMARY: 0 empty, 0 misaligned, 8 correct out of 8

(no misalignment detected in this run)
(Worker_TP0 pid=2784021) INFO 02-09 16:17:03 [multiproc_executor.py:707] Parent process exited, terminating worker
(Worker_TP3 pid=2784024) INFO 02-09 16:17:03 [multiproc_executor.py:707] Parent process exited, terminating worker
(Worker_TP2 pid=2784023) INFO 02-09 16:17:03 [multiproc_executor.py:707] Parent process exited, terminating worker
(Worker_TP1 pid=2784022) INFO 02-09 16:17:03 [multiproc_executor.py:707] Parent process exited, terminating worker
```

### CODE REVIEW

File: src/speculators/data_generation/custom_worker.py [hunk 1]
```diff
@@ -124,26 +135,61 @@ def _setup_hidden_states_capture(self, layer_ids: list[int]):
...
-        result = [
+
+        # Concatenate captured states from all scheduler iterations
+        concatenated_layers = [
             torch.cat(layer_tensors, dim=0) for layer_tensors in self._captured_states
         ]
-        # Clear intermediate storage after concatenating
+
+        # Slice and group by request
+        request_chunks: dict[str, list[list[torch.Tensor]]] = defaultdict(
```
Comment (rahul-tuli) [inline src/speculators/data_generation/custom_worker.py]: nit:
```suggestion
        request_chunks: defaultdict[str, list[list[torch.Tensor]]] = defaultdict(
```

**Review by rahul-tuli — APPROVED** [review]
LGTM! nice test!

File: src/speculators/data_generation/vllm_hidden_states_generator.py [hunk 2]
```diff
@@ -178,21 +180,22 @@ def _create_vllm_config(
         max_model_len: int,
         gpu_memory_utilization: float,
         tensor_parallel_size: int,
+        max_num_batched_tokens: int | None = None,
```
Comment (fynnsu) [inline src/speculators/data_generation/vllm_hidden_states_generator.py:183]: Seems like this isn't being used

Comment (shanjiaz) [reply src/speculators/data_generation/vllm_hidden_states_generator.py:183]: Yeah, good catch! It should be used in test the test. I might have removed an if statement by accident. Will update.
